test/Microsoft.IdentityModel.Tests/ReferenceXml.cs

```diff
@@ -561,6 +561,34 @@ public static SamlAssertionTestSet SamlAssertionMultiStatements
                 };
             }
         }
+
+        public static SamlAssertionTestSet SamlAssertionMultiStatements_DifferentSubject
+        {
+            get
+            {
+                SamlSubject subject = new SamlSubject(Default.NameIdentifierFormat, Default.NameQualifier, Default.Subject);
+                SamlAttributeStatement statement = new SamlAttributeStatement(subject, SamlAttributeSingleValue.Attribute);
+
+                return new SamlAssertionTestSet
+                {
+                    Assertion = new SamlAssertion(Default.SamlAssertionID, Default.Issuer, DateTime.Parse(Default.IssueInstant), SamlConditionsSingleCondition.Conditions, AdviceWithAssertionIDRef.Advice, new List<SamlStatement> { SamlAttributeStatementSingleAttribute.AttributeStatement, statement }),
+                    Xml = XmlGenerator.SamlAssertionXml(Convert.ToString(SamlConstants.MajorVersionValue), Convert.ToString(SamlConstants.MinorVersionValue), Default.SamlAssertionID, Default.Issuer, Default.IssueInstant, SamlConditionsSingleCondition.Xml, AdviceWithAssertionIDRef.Xml,
+                        string.Concat(SamlAttributeStatementSingleAttribute.Xml, XmlGenerator.SamlAttributeStatementXml(XmlGenerator.SamlSubjectXml(XmlGenerator.SamlNameIdentifierXml(Default.NameQualifier, Default.NameIdentifierFormat, Default.Subject), null), SamlAttributeSingleValue.Xml)),  null)
+                };
+            }
+        }
+
+        public static SamlAssertionTestSet SamlAssertionMultiStatements_DifferentStatementType
+        {
+            get
+            {
+                return new SamlAssertionTestSet
+                {
+                    Assertion = new SamlAssertion(Default.SamlAssertionID, Default.Issuer, DateTime.Parse(Default.IssueInstant), SamlConditionsSingleCondition.Conditions, AdviceWithAssertionIDRef.Advice, new List<SamlStatement> { SamlAttributeStatementSingleAttribute.AttributeStatement, SamlAuthenticationStatementValid.AuthenticationStatement }),
+                    Xml = XmlGenerator.SamlAssertionXml(Convert.ToString(SamlConstants.MajorVersionValue), Convert.ToString(SamlConstants.MinorVersionValue), Default.SamlAssertionID, Default.Issuer, Default.IssueInstant, SamlConditionsSingleCondition.Xml, AdviceWithAssertionIDRef.Xml, string.Concat(SamlAttributeStatementSingleAttribute.Xml, SamlAuthenticationStatementValid.Xml), null)
```

Comment:
I opened https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/661 to address that the Major and Minor version should be strings.